### PR TITLE
Add orbit skill

### PR DESCRIPTION
**Pull request by mamiksik**
An attempt to add orbiting skill. Not complete -> problem with drifting while orbiting.

The general idea for the skill is to move in the direction of the normal vector to the directional vector from ball to robot.

Acceleration is controlled by sinus of angle be desired with a double period. -> Full angle = Maximal acceleration. 0 angle = no acceleration.

### CODE REVIEW

**Review by Yuhanun — APPROVED** [review]
lgtm

File: src/stp/skills/Orbit.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    auto initX = cos(initialVelocity.toAngle()) * initialVelocity.length();
+    auto initY = sin(initialVelocity.toAngle()) * initialVelocity.length();
+
+    auto margin = stp::control_constants::ROBOT_RADIUS + 2 * stp::control_constants::BALL_RADIUS;
+    auto adjustDistance = directionVector.normalize().scale(info.getRobot()->get()->getDistanceToBall() - margin);
+
+    auto multiplier = -sin(0.5 * (directionVector - info.getAngle()).toAngle());
+
+    auto x = cos(normalAngle) * multiplier + cos(adjustDistance.toAngle()) * adjustDistance.length(); //- initX adjustment for initial velocity?;
+    auto y = sin(normalAngle) * multiplier + sin(adjustDistance.toAngle()) * adjustDistance.length(); //- initY adjustment for initial velocity?;
```
Comment (rolfvdhulst) [inline src/stp/skills/Orbit.cpp]: It's fine for code usually, but I would typically discourage 'overuse' of auto like this. These data types should be either double or Vector2; naming them explicitly helps a lot with readability IMO.

Comment (rolfvdhulst) [reply src/stp/skills/Orbit.cpp]: For really complicated types, it's fine to use auto, but here, I don't think it's necessary.

Comment (mamiksik) [reply src/stp/skills/Orbit.cpp]: I will add explicit types.

Comment (Yuhanun) [reply src/stp/skills/Orbit.cpp]: I don't necessarily agree with rolf here, though.

I think any type that's not a built-in type (aka arithmetic + ref/pointer qualified thereof), should be `auto`.

so yeah these variables should be explicitly typed, but if you get to std::vector etc i'd already auto them.

Comment (rolfvdhulst) [reply src/stp/skills/Orbit.cpp]: That's a style preference. Personally, I like auto only when it's sort-of clear from the context what the type is. For templated containers this often is the case with good naming and auto is fine, but I would personally still use the type for simple classes. When using long, chained calls, like happens here often, I don't want to spend 2 minutes trying to decipher what the last call is and what type it gives, so I'd be tempted to use auto a bit less often. For simple single function calls, auto becomes a lot more acceptable.

Comment (mamiksik) [reply src/stp/skills/Orbit.cpp]: ehh, don't we have a set code style?

Comment (rolfvdhulst) [reply src/stp/skills/Orbit.cpp]: No, impossible to enforce when the team changes every year. But for now you can do what you like, as long as you at least do not use auto for built-in types.

Comment (mamiksik) [reply src/stp/skills/Orbit.cpp]: I added types and amended the original commit. What about using a linter such as https://github.com/cpplint/cpplint?

Comment (rolfvdhulst) [reply src/stp/skills/Orbit.cpp]: I believe we already had a linter, but we never use/enforce it anymore these days :/

Comment (mamiksik) [reply src/stp/skills/Orbit.cpp]: I guess adding it to CI, for next year, would be a nice way to keep code consistent. btw: can we close/merge the PR if everything is fine?

Comment (Yuhanun) [reply src/stp/skills/Orbit.cpp]: We've tried adding it to CI but it's not like, perfect.

If we can get CI to format any file that changed then sure, feel free to set it up :)

Comment (mamiksik) [reply src/stp/skills/Orbit.cpp]: I think linter should not format code for you.  Also, not all linting rules are formating, aren't they? For example, PEP8 has a code style rule for if-else return. 

This code does not pass PEP8:
```python3
if x is Null:
    return "Fail"
else:
    return "Success"
``` 

This does:
```python
if x is Null:
    return "Fail"

return "Success"
```


=> I think CI should just disallow you to merge PR while linting check does not pass.

**Review by JaroKuiken — APPROVED** [review]
Ill merge

**Pull request merged**